src/opnsense/scripts/unbound/download_blacklists.py

```diff
@@ -58,10 +58,11 @@ def uri_reader(uri):
                 break
             else:
                 parts = (prev_chop + chop).split('\n')
-                if len(parts) > chop.find('\n'):
-                     prev_chop = parts.pop()
+                if parts[-1] != "\n":
+                    prev_chop = parts.pop()
                 else:
                     prev_chop = ''
+                    parts.pop()
```

Comment:
@kulikov-a this looks a bit weird, why do we need to remove the last line here?

Reply:
in line 65 parts array last memebr is "\n". do we need to yield it?

Reply:
well, you can chop it if it's empty. e.g.

```
("xxx\nyyy\naaa\n" + "iii\ntttt\n").split("\n")
```
vs.
```
("xxx\nyyy\naaa\n" + "iii\ntttt").split("\n")
```

in the last case it would chop "tttt"  if I'm not mistaken

Reply:
in first case we have full string, `parts[-1] == "\n"`, so `prev_chop = ''` and we dont need to yield `"\n"` ?
in second case we do not know if the whole line is received, so we pass this to `prev_chop` and `.pop()` it from `parts`.
line 65 can be can be safely removed.  work well with or without it. `\n` will be filtered at `tmp = line.split('#')[0].split()`.
just didn't want to yield "empty" data to `line`

Reply:
accidentally yielding an empty attribute likely doesn't mind, since the content is filtered anyway. I would remove the pop to be honest, it's less obvious what it's doing from the code.

Reply:
checked again. sorry. you're right. its '' not '\n' inf first case (my bad). still works (removing empty member) but absolutely useless.
removing..

Reply:
no problem, thanks for taking care of this